x-aeson/src/X/Data/Aeson.hs

```diff
@@ -19,6 +19,7 @@ import           Data.Text
 import           Data.Text.Encoding (decodeUtf8, encodeUtf8)
 import           Data.ByteString.Lazy (toStrict)
 
+import           Prelude -- AMP compatibility workaround
```

Comment:
Yeah, that seems more future-proof, given this trick kinda relies on an arcane implementation detail